### PR TITLE
Check only hostname for URL sources

### PR DESCRIPTION
**Pull request by pedromml**
This PR aims to solve https://github.com/OpenHistoricalMap/issues/issues/643
It uses the `URL()` constructor to check if the source is a URL and only uses the hostname on the regex checks.
New tests were also added to cover having a URL in the source string.

### CODE REVIEW

**Comment by pedromml** [comment]
This fix relies on the `URL()` constructor recognizing the entire source string as an URL, so if the user types in "I found this on https://abc.com/xyz-google", the check is going to default to comparing the entire string to the regexes. I tried separating the source string into words and checking each of them, but that would create a new problem: If a user typed in "Google drive", which should be ignored, the word "Google" would receive a warning. Should something be done abut these situations?

File: modules/validations/incompatible_source.js [hunk 1]
```diff
@@ -30,6 +30,12 @@ export function validationIncompatibleSource() {
 
     return entitySources
       .map(source => {
+        try {
+          const sourceURL = new URL(source);
+          source = sourceURL.hostname;
+        } catch (e) {
+          // Source string is not a url
```
Comment (1ec5) [inline modules/validations/incompatible_source.js:37]: Looks like this fallthrough will preserve the original passed-in value of `source`. 👍

**Review by 1ec5 — APPROVED** [review]
Thanks for fixing this issue. The tests look good, and I can’t think of any other edge cases that would pose a problem here.

**Comment by 1ec5** [comment]
> This fix relies on the `URL()` constructor recognizing the entire source string as an URL, so if the user types in "I found this on https://abc.com/xyz-google", the check is going to default to comparing the entire string to the regexes. I tried separating the source string into words and checking each of them, but that would create a new problem: If a user typed in "Google drive", which should be ignored, the word "Google" would receive a warning. Should something be done abut these situations?

You have a point that the rule will only catch either problematic URLs in isolation or phrases that aren’t URLs, but not URLs embedded in other text. I suspect this will be unlikely for the particular sources we’re flagging. The goal is to head off inadvertent mistakes or casual, well-meaning usage of problematic sources, but there are plenty of ways to intentionally get around the warning, such as not entering a source at all.

**Comment by 1ec5** [comment]
For future reference: you’ll generally want to avoid posting a pull request from a branch with the same name as the target branch. In this case, you’d want to create a new branch specific to the bug fix. PRing staging into staging will give you a harder time keeping the branch up-to-date as this upstream repository changes. Once this PR lands, you can delete your branch and fetch it again.
